### ISSUE TITLE
Docs feat request: Modules, Optimizers, LMs, RMs should have links back to the source code somewhere on the page

### ISSUE THREAD

**Issue by isaacbmiller** [stanfordnlp/dspy#1709]
This is an opinion, but it would be great if the different signatures had links back to source code.

E.g. https://dspy-docs.vercel.app/deep-dive/modules/react/ should link back to https://github.com/stanfordnlp/dspy/blob/main/dspy/predict/react.py

**Comment by isaacbmiller** [comment]
cc @krypticmouse thoughts?

**Comment by krypticmouse** [comment]
Good idea, we might have API refs back at some point and when that happens we can probably add this in them?

**Comment by okhat** [comment]
cc'ing also @ammirsm
